### ISSUE TITLE
When clicking a non-preloaded tab from toolbar, willEnter and didEnter are not fired

### ISSUE THREAD

**Issue by abervoet** [zyra/ionic2-super-tabs#185]
Hello again,

### Scenario
When you click on the super-tabs toolbar to navigate to a tab component that has not been yet loaded (never visited  and never been a neighbour from the active tab).

- the target tab component is instanciated (`ctor`), 
- `willLoad` and `didLoad `are fired,
-  **`willEnter `and `didEnter` are not fired**. 

This is an issue when you do want to execute some tab's component's code in `willEnter` or `didEnter`.
The issue occurs only when clicking the toolbar, not when doing regular tabs sliding.

### Code example
[https://github.com/abervoet/ionic2-super-tabs-issue](https://github.com/abervoet/ionic2-super-tabs-issue)

Enter the HomePage that contains `<super-tabs>` with 4 `<super-tab>` -> tab1 is active, neighbour tab2 component is pre-loaded.
Then click on toolbar to tab3 or tab4.

See [screenshot ](https://imgur.com/a/at2zg) of the use case.

### Versions
```
    "@angular/common": "4.4.3",
    "@angular/core": "4.4.3",
    "ionic-angular": "3.7.1",
    "ionic2-super-tabs": "^4.1.1",
```

### Related code (generated js)
In [SuperTabs.onTabChange()#511](https://github.com/zyra/ionic2-super-tabs/blob/master/src/components/super-tabs.ts#L511) -> `this.fireLifecycleEvent(['willEnter', 'didEnter']);`.
The method contains the line: `activeView = this.getActiveTab().getActive();` which always returns `undefined` in the described case because the component's instanciation has not been made at this point.
```
SuperTabs.prototype.onTabChange = function (index) {
        index = Number(index);
        if (index === this.selectedTabIndex) {
            this.tabSelect.emit({
                index: index,
                id: this._tabs[index].tabId,
                changed: false
            });
            return;
        }
        if (index <= this._tabs.length) {
            console.log('Changing tab');
            var currentTab = this.getActiveTab();
            var activeView = currentTab.getActive();
            if (activeView) {
                console.log('Firing leave events');
                activeView._willLeave(false);
                activeView._didLeave();
            }
            this.selectedTabIndex = index;
            this.linker.navChange(DIRECTION_SWITCH);
            this.refreshTabStates();
            activeView = this.getActiveTab().getActive(); // <---- undefined when clicking the toolbar to a tab component that has not been instanciated yet)
            if (activeView) {
                console.log('Firing enter events');
                activeView._willEnter();
                activeView._didEnter();
            }
            this.tabSelect.emit({
                index: index,
                id: this._tabs[index].tabId,
                changed: true
            });
        }
    };
```

**Comment by ihadeed** [comment]
Hi,

This was fixed in 4.1.3

**Comment by abervoet** [comment]
Great, sorry for not noticing the new version.
Verified
